### PR TITLE
Bug 1820740 - Add UI tests coverage for HTTPS-only mode

### PR DESCRIPTION
**Pull request by sv-ohorvath**
Other changes needed on this PR:
- Test classes related to the Settings sub-menus needed a refactor to reflect the current layout and to be better organized.
- Updated the androidTest/AppRequestInterceptor.kt to match the [production code](https://github.com/mozilla-mobile/firefox-android/blob/main/fenix/app/src/main/java/org/mozilla/fenix/AppRequestInterceptor.kt), otherwise, it will not serve the correct error pages.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1820773
https://bugzilla.mozilla.org/show_bug.cgi?id=1820740
https://bugzilla.mozilla.org/show_bug.cgi?id=1821024

### CODE REVIEW

**Review by AndiAJ — APPROVED** [review]
👍🏽 LGTM !

However, I'm not sure about the `AppRequestInterceptor` changes, perhaps @Mugurell is kind enough to review them ?

**Comment by Mugurell** [comment]
> > 👍🏽 LGTM !
> > However, I'm not sure about the `AppRequestInterceptor` changes, perhaps @Mugurell is kind enough to review them ?
> 
> Sure, but I only copied over code from https://github.com/mozilla-mobile/firefox-android/blob/main/fenix/app/src/main/java/org/mozilla/fenix/AppRequestInterceptor.kt

Thank you for the ping. That change LGTM.
Looks like we backported to use in tests the behavior updated in the meantime for the app.
I see that the only difference needed in tests is documented nicely - https://github.com/mozilla-mobile/firefox-android/blob/acc0d9638e9e97f55fb25fede40c3e1176dd726b/fenix/app/src/androidTest/java/org/mozilla/fenix/AppRequestInterceptor.kt#L132-L133 which is great. Maybe as a small improvement, a similar comment could be left also in the production code, for when developers modify that class, to know that any change must also be added to the class used in tests.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/LoginsTest.kt [hunk 1]
```diff
@@ -52,6 +48,42 @@ class LoginsTest {
         mockWebServer.shutdown()
     }
 
+    // Tests only for initial state without signing in.
+    // For tests after singing in, see SyncIntegration test suite
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/LoginsTest.kt:52]: typo: `signing`

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/LoginsTest.kt [hunk 2]
```diff
@@ -52,6 +48,42 @@ class LoginsTest {
...
+            // Necessary to scroll a little bit for all screen sizes
+            TestHelper.scrollToElementByText("Logins and passwords")
+        }.openLoginsAndPasswordSubMenu {
+            verifyDefaultView()
+            verifyDefaultValueAutofillLogins(TestHelper.appContext)
+            verifyDefaultValueExceptions()
+        }.openSavedLogins {
+            verifySecurityPromptForLogins()
+            tapSetupLater()
+            // Verify that logins list is empty
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/LoginsTest.kt:67]: We should have the check with the call in-place but commented out.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/LoginsTest.kt [hunk 3]
```diff
@@ -52,6 +48,42 @@ class LoginsTest {
...
+            // Issue #7272 nothing is shown
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/LoginsTest.kt:68]: Is this now fixed? I checked https://github.com/mozilla-mobile/fenix/issues/7272 and it seems fixed. If not, we should be linking to Bugzilla now.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsDeleteBrowsingDataTest.kt [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+        homeScreen {
+            verifyExistingTopSitesTabs(pocketTopArticles)
+        }.openTopSiteTabWithTitle(pocketTopArticles) {
+            waitForPageToLoad()
+        }.openTabDrawer {
+        }.openNewTab {
+        }.submitQuery("about:cache") {
+            // disabling wifi to prevent downloads in the background
+            setNetworkEnabled(enabled = false)
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsDeleteBrowsingDataTest.kt:239]: Does this work on Firebase Test Lab?

**Review by AaronMT — APPROVED** [review]
See comments, no blockers. Assuming these run on FTL fine.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsDeleteBrowsingDataTest.kt (hunk 4 above)
Comment (sv-ohorvath) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsDeleteBrowsingDataTest.kt:239]: yes, it's not a new test, I just moved it.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/LoginsTest.kt (hunk 3 above)
Comment (sv-ohorvath) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/LoginsTest.kt:68]: I'll check it out

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/LoginsTest.kt (hunk 1 above)
Comment (sv-ohorvath) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/LoginsTest.kt:52]: done

**Pull request merged**